roxie/udplib/udptrr.cpp

```diff
@@ -643,46 +647,85 @@ class CReceiveManager : implements IReceiveManager, public CInterface
         #else
             adjustPriority(2);
         #endif
-            DataBuffer *b = NULL;
             started.signal();
             unsigned lastOOOReport = 0;
             unsigned lastPacketsOOO = 0;
+            unsigned timeout = 5000;
+            DataBuffer *b = nullptr;
             while (running) 
             {
                 try 
                 {
                     unsigned int res;
-                    b = bufferManager->allocate();
-                    while (true)
+                    bool dataAvail = receive_socket->wait_read(timeout);
+                    if (dataAvail)
                     {
-                        receive_socket->read(b->data, 1, DATA_PAYLOAD, res, 5);
+                        if (!b)
+                            b = bufferManager->allocate();
+                        receive_socket->readtms(b->data, 1, DATA_PAYLOAD, res, 0);
                         if (res!=sizeof(UdpRequestToSendMsg))
```

Comment:
Is a data send always guaranteed to be larger than sizeof(udpRequestToSendMsg) ?

Reply:
Yes - and I think I put in a static assert to confirm that.